sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -3,13 +3,15 @@ package io.sentry.android.okhttp
 import io.sentry.Breadcrumb
 import io.sentry.HubAdapter
 import io.sentry.IHub
+import io.sentry.ISpan
 import io.sentry.SpanStatus
 import java.io.IOException
 import okhttp3.Interceptor
 import okhttp3.Response
 
 class SentryOkHttpInterceptor(
-    private val hub: IHub = HubAdapter.getInstance()
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val spanCustomizer: (span: ISpan) -> Unit = { }
```

Comment:
what's about using a name that sort of matches our beforeSend and beforeBreadcrumb callbacks?
eg `beforeSpan`? cc @bruno-garcia 

I believe this could also be an interface already available in the sentry package, it may be reused in other integrations, I'd not add to options though, wdyt?

Reply:
2nd thing is that using `Unit` isn't the best definition if using it on Java, but as OkHttp is already written in Kotlin, I guess it's just fine.

Reply:
Agreed with naming. I think `Unit` can stay.